### DISCUSSION TITLE
2022-07-14のミーティングアジェンダ

### DISCUSSION THREAD

**Discussion by azu** [asciidwango/js-primer#1449] [Meeting]
2022-07-14のミーティングアジェンダ

**Comment by azu** [comment]
[非同期の章の改善案を募集 · Discussion #1444 · asciidwango/js-primer](https://github.com/asciidwango/js-primer/discussions/1444)

**Reply by azu** [reply]
非同期といったときに並列と想像してしまう人がいるので、JSだと並列だよというのだけわかるような感じ。
並行の詳細はあんまり深掘りしない

**Reply by azu** [reply]
PromiseとAsyncだけを知りたいというニーズ に対してちょっと量が多い。

エラーファーストコールバック

**Reply by azu** [reply]
## 非同期の章

- Promise
- [コラム] エラーファーストコールバック
- Async

## ユースケース CLI

- `fs/promises`を基本にして
  - `*/promises` 
- コラムでエラーファーストコールバック

**Reply by azu** [reply]
<img width="547" alt="image" src="https://user-images.githubusercontent.com/19714/178962286-01e8c339-bd8a-462b-8f2a-30d224569da5.png">

- 非同期処理と例外処理
- エラーファーストコールバック

をPromiseの解説した後に移す。
Promiseを使わない時の例外処理的な話として一つにまとまる?

**Reply by azu** [reply]
Promiseは新しいものではなくなった。

---

## [コラム] エラーファーストコールバック

Promiseの説明後に、Promise以前はエラーファーストコールバックが使われていた。
Node.jsではこれがコアでも広く使われています。

**Reply by azu** [reply]
> 非同期処理:コールバック/Promise/Async Function

を

> 非同期処理: Promise/Async Function

に変える

**Reply by azu** [reply]
- 非同期処理はその処理が終わるのを待つ前に次の処理を評価すること
- 非同期処理であってもメインスレッドで実行されることがある
- Promiseは、ES2015で導入された非同期処理を扱うビルトインオブジェクト
- Async Functionは、ES2017で導入された非同期処理を扱う構文
- Async FunctionはPromiseの上に作られた構文であるため、Promiseと組み合わせて利用する

**Reply by azu** [reply]
[Child process | Node.js v18.7.0 Documentation](https://nodejs.org/api/child_process.html) には/promiseがないんだ。

**Comment by azu** [comment]
[`function` キーワードと関数式のうち推奨のものがあればそれを、ない場合はその旨明記してほしい · Issue #1354 · asciidwango/js-primer](https://github.com/asciidwango/js-primer/issues/1354)

**Reply by lacolaco** [reply]
ユースケースなどのサンプルコード内で表記ゆれのような感じになってないかは確認しておいてよさそう

**Reply by lacolaco** [reply]
なさそう　（そもそも関数宣言がほとんどない）

**Reply by azu** [reply]
https://jsprimer.net/basic/date/#instance-method
https://jsprimer.net/basic/module/#named-export-import

この本ではどっちを使います。

**Reply by azu** [reply]
`const fn = () => {}` or `function fn(){}`

- 実用はconst
- 文章的にはfunction

**Reply by azu** [reply]
今は function になってる。

**Reply by azu** [reply]
とりあえずfunctionのままにする

**Comment by azu** [comment]
`node:*` prefixのモジュール名を使うようにしたい。

[Node.js 18 Introduces Prefix-Only Core Modules](https://fusebit.io/blog/node-18-prefix-only-modules/)

**Reply by azu** [reply]
https://github.com/asciidwango/js-primer/issues/1437
Node 16+から利用できる?

**Reply by azu** [reply]
https://nodejs.org/en/about/releases/
18は10月25日にリリースされる。

**Reply by azu** [reply]
https://jsprimer.net/use-case/nodecli/refactor-and-unittest/#create-env

devDependencies ぐらい、mochaである必要性。

https://nodejs.org/api/test.html
はまだexperimental

**Reply by azu** [reply]
- mochaをアップデート
- `node:assert`
- `node:test`は使わない

**Comment by azu** [comment]
スケジュール

**Reply by azu** [reply]
10月末に原稿がfixされている

**Comment by azu** [comment]
ajaxのevent listener

**Reply by azu** [reply]
onclick="xxx" をやめて、addEventListnerにするか

**Reply by azu** [reply]
これは次回にSKIP

**Comment by azu** [comment]
次回 8月18日(木) 19:00
